Tests fail to build against openssl 1.1.0.f

Here is the error, with the complete command, running on rustc 1.20.0-nightly (9475ae477 2017-07-11):
```
error: linking with `cc` failed: exit code: 1              
  |                                                        
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.alloc-allocator.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.alloc-arc.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.alloc-borrow.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.alloc-heap.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.alloc-raw_vec.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.alloc-slice.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.alloc-str.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.alloc-vec.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.core-any.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.core-array.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.core-cmp.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.core-convert.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.core-fmt-builders.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.core-fmt.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.core-nonzero.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.core-ptr.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.core-result.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.core-slice.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.curl-easy.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.smoke.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.smoke-__test.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-collections-hash-table.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-io-error.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-net-addr.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-net-tcp.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-net.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-panic.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-panicking.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-path.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-process.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-sync-mutex.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-sys_common-backtrace.volatile.o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.std-thread.volatile.o" "-o" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/smoke-9e4ed3045f2c9c03.crate.allocator.o" "-Wl,--gc-sections" "-pie" "-nodefaultlibs" "-L" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps" "-L" "/usr/lib" "-L" "/usr/lib" "-L" "/usr/lib" "-L" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/build/backtrace-sys-5cb185e52a30e43c/out/.libs" "-L" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/libcurl-c2ddb7519ea99d1f.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libtest-f810631b1b7e6a95.rlib" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/libcurl_sys-366b61fdb4cd2acb.rlib" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/liblibz_sys-afc6cab14c210a41.rlib" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/libopenssl_sys-682483b06fd60e73.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgetopts-1b892f4587025614.rlib" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/liblibc-10310e83a4a4a93b.rlib" "/home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/libopenssl_probe-e71eb66f2bb158bd.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libterm-15040a509e79d2c3.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-e18aff57b2094f45.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-3dafff1bd2cd5f82.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc_jemalloc-57bcc55715463b4b.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-382b1b0334de8b04.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc_system-3b051e112a4598f3.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-ec14a360d4d3acaf.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-33f7e2439faf585a.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_unicode-66d49f9f7e114d5c.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librand-6c8f8c6ff7d1a169.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-2f52d9d1a3ebf41c.rlib" "/home/linkmauve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-c5920ad7de4f466c.rlib" "-Wl,-Bdynamic" "-l" "curl" "-l" "z" "-l" "ssl" "-l" "crypto" "-l" "util" "-l" "dl" "-l" "rt" "-l" "pthread" "-l" "pthread" "-l" "gcc_s" "-l" "c" "-l" "m" "-l" "rt" "-l" "pthread" "-l" "util"                                           
  = note: /home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/libopenssl_sys-682483b06fd60e73.rlib(openssl_sys-682483b06fd60e73.1.o): In function `openssl_sys::set_id_callback':                                                          
          /home/linkmauve/data/cargo/registry/src/github.com-1ecc6299db9ec823/openssl-sys-0.7.17/src/lib.rs:544: undefined reference to `CRYPTO_set_id_callback'                                                                               
          /home/linkmauve/dev/rust/tokio-socks5/target/debug/deps/libopenssl_sys-682483b06fd60e73.rlib(openssl_sys-682483b06fd60e73.2.o): In function `openssl_sys::init::{{closure}}':                                                        
          /home/linkmauve/data/cargo/registry/src/github.com-1ecc6299db9ec823/openssl-sys-0.7.17/src/lib.rs:517: undefined reference to `SSL_library_init'                                                                                     
          /home/linkmauve/data/cargo/registry/src/github.com-1ecc6299db9ec823/openssl-sys-0.7.17/src/lib.rs:518: undefined reference to `SSL_load_error_strings'                                                                               
          /home/linkmauve/data/cargo/registry/src/github.com-1ecc6299db9ec823/openssl-sys-0.7.17/src/lib.rs:519: undefined reference to `OPENSSL_add_all_algorithms_noconf'                                                                    
          /home/linkmauve/data/cargo/registry/src/github.com-1ecc6299db9ec823/openssl-sys-0.7.17/src/lib.rs:521: undefined reference to `CRYPTO_num_locks'                                                                                     
          /home/linkmauve/data/cargo/registry/src/github.com-1ecc6299db9ec823/openssl-sys-0.7.17/src/lib.rs:531: undefined reference to `CRYPTO_set_locking_callback'                                                                          
          collect2: error: ld returned 1 exit status
```

I’m not sure which dependency is trying to link against it though.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,4 +11,4 @@ env_logger = "0.3"
 trust-dns = { version = "0.9.2", default-features = false }
 
 [dev-dependencies]
-curl = "0.3"
+curl = "0.4"

